openmmtools/testsystems.py

```diff
@@ -350,6 +357,19 @@ def positions(self):
         del self._positions
 
     @property
+    def topology(self):
```

Comment:
Currently, `testsystem._topology` is mutable, `testsystem.topology` returns an immutable copy.

We do the same with `testsystem.system`.  

I'm happy to go either way on this.  In particular, I'd love to go with whichever behavior is the least surprising to users/developers.